internal/environment_writer.go

```diff
@@ -37,6 +37,11 @@ func (w EnvironmentWriter) Write(path string, environment map[string]string) err
 
 	for key, value := range environment {
 		f := filepath.Join(path, key)
+
+		if err := os.MkdirAll(filepath.Dir(f), 0755); err != nil {
```

Comment:
Should this be outside the for loop? Isn't the directory for all the env vars the same?

Reply:
Normally, but when you have an environment variable that's specific to a process type, there is a subfolder. That means key here will look like this: "health-check/THC_PATH.default". It presently fails if you try to do this because the process type subfolder isn't created.

Reply:
Can we add this as a comment as well?

Reply:
Done.